src/ArticleController.m

```diff
@@ -463,8 +458,80 @@ -(void)selectFolderAndArticle:(NSInteger)folderId guid:(NSString *)guid
  */
 -(void)reloadArrayOfArticles
 {
+    Article * article = mainArticleView.selectedArticle;
+
+	reloadArrayOfArticlesSemaphor++;
+	// add a progress indicator
+	__block NSProgressIndicator * progressIndicator = [[NSProgressIndicator alloc] initWithFrame:mainArticleView.mainView.visibleRect];
+	progressIndicator.style = NSProgressIndicatorSpinningStyle;
+	[progressIndicator setDisplayedWhenStopped:NO];
+	[mainArticleView.mainView addSubview:progressIndicator];
```

Comment:
It is an anti-pattern to modify `mainArticleView.mainView` outside the view/viewController. I will soon fork your branch and submit a PR to change this. 
It should be implemented something like this:
```objective-c
[mainArticleView.mainView createProgressIndicator]; // creates a progress indicator and adds to subview
[mainArticleView.mainView startProgressIndicator]; // starts the progress indicator
```